api/src/main/java/org/apache/iceberg/ExpireSnapshots.java

```diff
@@ -118,4 +118,16 @@ public interface ExpireSnapshots extends PendingUpdate<List<Snapshot>> {
    * @return this for method chaining
    */
   ExpireSnapshots cleanExpiredFiles(boolean clean);
+
+  /**
+   * Allows expiration of unreachable metadata, such as partition spec as part of the operation.
+   *
+   * @param clean setting this to true will remove metadata(such as partition spec) that is no
+   *     longer reachable by any snapshot
+   * @return this for method chaining
```

Comment:
```
  /**
   * Allows expiration of unreachable table layout metadata, such as partition specs as part of the operation.
   *
   * @param setting this to true will remove table layout metadata that is no
   *     longer reachable by any snapshot
   * @return this for method chaining
```

Reply:
Basically don't think we need to mention the "such as partition spec" twice, I think once at the top level and clarifying that we're talking about layout metadata like the spec, and in the future the schema should be sufficient imo.